### PR TITLE
[Gekidou] Typings & PostMetadata structure

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
* Removed unnecessary `Raw` typings
* Fixed the typings everywhere
* Made it so post only contains one PostMetadata
* Cleanup Android MainApplication.java
* Can't remember what else :P

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/helpers/api/team.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DEFAULT_LOCALE} from '@i18n';
+
+export const selectDefaultTeam = (teams: Team[], locale = DEFAULT_LOCALE, userTeamOrderPreference = '', primaryTeam = '') => {
+    let defaultTeam;
```
Comment (avinashlng1080) [inline app/helpers/api/team.ts:7]: Do we also add unit test to helper functions ?

**Review by avinashlng1080 — COMMENTED** [review]
Please see below comments.

File: app/database/operator/app_data_operator/index.test.ts [hunk 2]
```diff
@@ -88,49 +85,4 @@ describe('** APP DATA OPERATOR **', () => {
             prepareRecordsOnly: false,
         });
     });
-
-    it('=> HandleServers: should write to SERVERS table', async () => {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/database/operator/app_data_operator/index.test.ts:92]: cause it was never used.. we create servers with the manager

File: types/api/users.d.ts [hunk 3]
```diff
@@ -18,25 +18,30 @@ type UserNotifyProps = {
     user_id?: string;
 };
 
+interface UserProps {
```
[the remark this answers is not in the document]
Comment (enahum) [reply types/api/users.d.ts:21]: good catch

File: app/helpers/api/preference.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export function getPreferenceValue(preferences: PreferenceType[], category: string, name: string, defaultValue: unknown = '') {
+    const pref = preferences.find((p) => p.category === category && p.name === name);
+
+    return pref?.value || defaultValue;
+}
+
+export function getPreferenceAsBool(preferences: PreferenceType[], category: string, name: string, defaultValue = false) {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/helpers/api/preference.ts:10]: nope, is a boolean not a string

File: app/database/operator/app_data_operator/transformers/index.ts [hunk 5]
```diff
@@ -63,34 +65,3 @@ export const transformGlobalRecord = ({action, database, value}: TransformerArgs
...
 };
 
-/**
- * transformServersRecord: Prepares a record of the APP database 'Servers' table for update or create actions.
- * @param {TransformerArgs} operator
- * @param {Database} operator.database
- * @param {RecordPair} operator.value
- * @returns {Promise<Model>}
- */
-export const transformServersRecord = ({action, database, value}: TransformerArgs) => {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/database/operator/app_data_operator/transformers/index.ts:73]: Not removed from the App Database, but this functions are not used

File: app/helpers/api/team.ts (hunk 1 above)
Comment (enahum) [reply app/helpers/api/team.ts:7]: Not yet.. this is part of the entry point logic that I'm working on.

**Review by avinashlng1080 — APPROVED** [review]
LGTM

**Pull request merged**